src/server/system_services/bucket_server.js

```diff
@@ -513,6 +513,18 @@ function _validate_s3_policy(policy, bucket_name) {
                 throw new RpcError('MALFORMED_POLICY', 'Policy has invalid action', { detail: action });
             }
         }
+        if (statement.Condition) {
+            for (const condition of Object.values(statement.Condition)) {
+                for (const condition_key of Object.keys(condition)) {
+                    // some condition keys have arguments in their names.(e.g. s3:ExistingObjectTag/<key>)
+                    // parse to get only the condition key itself
+                    const key_parts = condition_key.split("/");
+                    if (!SUPPORTED_BUCKET_POLICY_CONDITIONS.includes(key_parts[0])) {
+                        throw new RpcError('MALFORMED_POLICY', 'Policy has invalid condition key', { detail: condition_key });
```

Comment:
Should we return "malformed policy" in case of a condition we don't support? Maybe we should at least add something like this:
```suggestion
                        throw new RpcError('MALFORMED_POLICY', 'Policy has invalid or unsupported condition key', { detail: condition_key });
```
WDYT?